consent-ui/src/app/ais/sca-result-page/sca-result-page.component.ts

```diff
@@ -0,0 +1,74 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {ActivatedRoute, Router} from "@angular/router";
+import {StubUtil} from "../common/stub-util";
+import {ConsentAuthorizationService} from "../../api";
+import {Subscription} from "rxjs";
+import {ApiHeaders} from "../../api/api.headers";
+import {SessionService} from "../../common/session.service";
+
+@Component({
+  selector: 'consent-app-sca-result-page',
+  templateUrl: './sca-result-page.component.html',
+  styleUrls: ['./sca-result-page.component.scss']
+})
+
+export class ReportScaResultComponent implements OnInit, OnDestroy {
+  public static ROUTE = 'sca-result';
+
+  reportScaResultForm: FormGroup;
+  private authorizationSessionId: string;
+  private redirectCode: string;
+  private subscriptions: Subscription[] = [];
+
+  constructor(private sessionService: SessionService,
+              private consentAuthorizationService: ConsentAuthorizationService,
+              private activatedRoute: ActivatedRoute,
+              private router: Router,
+              private formBuilder: FormBuilder) {
+  }
+
+  ngOnInit() {
+    this.reportScaResultForm = this.formBuilder.group({
+      tan: ['', Validators.required]
+    });
+
+    this.authorizationSessionId = this.activatedRoute.parent.snapshot.paramMap.get("authId");
+
+    this.consentAuthorizationService.authUsingGET(
+      this.authorizationSessionId,
+      this.sessionService.getRedirectCode(this.authorizationSessionId),
+      'response'
+    ).subscribe(res => {
+      this.redirectCode = res.headers.get(ApiHeaders.REDIRECT_CODE);
+      this.sessionService.setRedirectCode(this.authorizationSessionId, res.headers.get(ApiHeaders.REDIRECT_CODE));
+      console.log(res)
+    })
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(s => s.unsubscribe());
+  }
+
+  submit(): void {
+    this.subscriptions.push(
```

Comment:
Note: While it is always good to unsubscribe, HttpClient observables unsubscribe themselves

Reply:
done. 
subscriptions were deleted